lib/gen_stage.ex

```diff
@@ -2313,16 +2339,30 @@ defmodule GenStage do
     stage
   end
 
-  defp buffer_events(events, %{buffer: buffer, buffer_keep: keep} = stage) do
+  defp buffer_events(
+         events,
+         %{
+           mod: mod,
+           buffer: buffer,
+           buffer_keep: keep,
+           mod_handle_buffered: mod_handle_buffered,
+           state: state
+         } = stage
+       ) do
     {buffer, excess, perms} = Buffer.store_temporary(buffer, events, keep)
 
-    case excess do
-      0 ->
-        :ok
+    if mod_handle_buffered do
+      buffered_count = Buffer.estimate_size(buffer)
+      mod.handle_buffered(buffered_count, excess, state)
+    else
+      case excess do
```

Comment:
this should happen even if mod_handle_buffered is true right?

Reply:
This callback is basically made to replace this log, it gives us the ability to do a log or something else.

Reply:
For context this is the super mega spammy log that kept us using `lager` so we could filter it out, since it was causing issues when we were load shedding and dropping from the buffer.

Reply:
oh, I forgot briefly this was the project that we forked. so we don't super need to worry about how other people use it